crates/router/src/connector/adyen.rs

```diff
@@ -961,7 +975,7 @@ impl
                 field_name: "currency",
             })?,
         };
-        let amount = match data.request.amount {
+        let amount = match data.request.minor_amount {
             Some(amount) => amount,
             None => Err(errors::ConnectorError::MissingRequiredField {
                 field_name: "amount",
```

Comment:
can we convert after matching, since we are going to check request amount with response amount. we shouldn't check directly

Reply:
This is also from your commit